mdpow/workflows/base.py

```diff
@@ -0,0 +1,180 @@
+
+        locations = []
+
+        reg_compile = re.compile('FEP')
+        for dirpath, dirnames, filenames in os.walk(parent_directory):
+            result = [dirpath.strip() for dirname in dirnames if  reg_compile.match(dirname)]
+            if result:
+                locations.append(result[0])
+
+        resnames = []
+
+        for loc in locations:
+            res_temp = loc.strip().split('/')
+            resnames.append(res_temp[-1])
+
+        project_paths = pd.DataFrame(
+        {
+            'molecule': resnames,
+            'resname': resnames,
+            'path': locations
+        }
+    )
+        if csv_save_dir is not None:
+            
+            project_paths.to_csv(f'{csv_save_dir}/project_paths.csv', index=False)
+            logger.info(f'project_paths saved under {csv_save_dir}')
+
+        else:
+            current_directory = os.getcwd()
+            project_paths.to_csv('project_paths.csv', index=False)
+            logger.info(f'project_paths saved under {current_directory}')
+
+    elif csv is not None:
+        locations = pd.read_csv(csv)
+        project_paths = locations.sort_values(by=['molecule', 'resname', 'path']).reset_index(drop=True)
+
+    return project_paths
+
+def automated_project_analysis(project_paths, ensemble_analysis, **kwargs):
+    """Takes a :class:`pandas.DataFrame` created by :func:`~mdpow.workflows.base.project_paths`
+       and iteratively runs the specified :class:`~mdpow.analysis.ensemble.EnsembleAnalysis`
+       for each of the projects by running the associated automated workflow
+       in each project directory returned by :func:`~mdpow.workflows.base.project_paths`.
+
+       Compatibility with more automated analyses in development.
+
+       :keywords:
+
+       *project_paths*
+           :class:`pandas.DataFrame` that provides paths to MDPOW projects
+
+       *ensemble_analysis*
+           name of the :class:`~mdpow.analysis.ensemble.EnsembleAnalysis`
+           that corresponds to the desired automated workflow module
+
+       *kwargs*
+           keyword arguments for the supported automated workflows,
+           currently - :func:`~mdpow.workflows.dihedrals.automated_dihedral_analysis`
+
+           .. autodata:: mdpow.workflows.dihedrals.automated_dihedral_analysis
+
+       .. seealso:: :class:`~mdpow.analysis.dihedral.DihedralAnalysis`, :mod:`mdpow.workflows.dihedrals`
+       .. rubric:: Example
+
+       Typical Workflow::
+
+           project_paths = project_paths(parent_directory='/foo/bar/MDPOW_projects')
+           automated_project_analysis(project_paths, ensemble_analysis='DihedralAnalysis', **kwargs)
+
+    """
+
+    try:
+        for row in project_paths.itertuples():
+            molname = row.molecule
+            resname = row.resname
+            dirname = row.path
+
+            logger.info(f'starting {molname}')
+
+            registry.registry[ensemble_analysis](dirname=dirname, resname=resname, molname=molname, **kwargs)
```

Comment:
Wrap _only_ this line in try/except.

The smaller the code that you're bracketing, the more precisely you are still able to locate other errors.